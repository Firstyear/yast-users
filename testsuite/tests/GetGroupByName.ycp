{

    include "testsuite.ycp";
    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);

    import "Users";

    Users::groups_by_name ["local" ] = $[
        "users": $[
            "groupname": "users", "userlist":"", "more_users": "hh",
            "type": "local"
        ],
        "audio": $[
            "groupname": "audio", "userlist":"hh", "more_users": "",
            "type": "system"
        ]
    ];

    Users::groups_by_name ["ldap" ] = $[
        "users": $[
            "groupname": "users", "userlist":"", "more_users": "u1,u2",
            "type": "ldap"
        ]
    ];

    DUMP ( "first group with this name:" );

    TEST(``(Users::GetGroupByName ("users", "not_known")), [], 0);

    DUMP ( "non-existent group:" );

    TEST(``(Users::GetGroupByName ("video", "local")), [], 0);

    DUMP ( "type specified:" );

    TEST(``(Users::GetGroupByName ("users", "local")), [], 0);

}
