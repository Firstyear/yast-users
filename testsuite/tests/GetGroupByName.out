Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Execute	.target.bash_output "locale -k charmap" $[]
Dump	first group with this name:
Return	$["groupname":"users", "more_users":"u1,u2", "type":"ldap", "userlist":""]
Dump	non-existent group:
Log	Group video doesn't exist!
Return	$[]
Dump	type specified:
Return	$["groupname":"users", "more_users":"hh", "type":"local", "userlist":""]
